### PR TITLE
Small fixes and improved parsing for models in the MMCIF parser, inserti...

### PR DESCRIPTION
**Pull request by biomadeira**
...on codes and line_counter.

I was having a problem in parsing some multiple model mmCIF structures. Particularly when the length of the models were variable. I ended up making a few additional changes for consistency mostly. 

Kind regards,
Fábio

### CODE REVIEW

**Comment by JoaoRodrigues** [comment]
Looks good,  is there a multi model unit test in the suite? Otherwise we could add a small one.

**Comment by biomadeira** [comment]
@JoaoRodrigues I extended testModel() in test_MMCIF.py with just another parsing step using one of structures which was failing. Both the structures pass now so I would assume I didn't break anything else in the process... Just for sanity I ran the other PDB tests and it seems to be fine. All fine in Travis as well.
Thanks

File: Bio/PDB/MMCIF2Dict.py [hunk 1]
```diff
@@ -47,6 +49,8 @@ def __init__(self, filename):
                     self[key] = token
                     key = None
 
+    # Private methods
```
Comment (peterjc) [inline Bio/PDB/MMCIF2Dict.py]: Methods with a single leading underscore are by Python convention private, so the comment is redundant (but harmless)

File: Bio/PDB/StructureBuilder.py [hunk 2]
```diff
@@ -196,15 +195,15 @@ def init_atom(self, name, coord, b_factor, occupancy, altloc, fullname,
                                   % (duplicate_fullname, fullname,
                                      self.line_counter),
                                   PDBConstructionWarning)
-        atom = self.atom = Atom(name, coord, b_factor, occupancy, altloc,
-                                fullname, serial_number, element)
+        self.atom = Atom(name, coord, b_factor, occupancy, altloc,
+                         fullname, serial_number, element)
```
Comment (peterjc) [inline Bio/PDB/StructureBuilder.py]: Is the good reason for adding a new public attribute ``.atom`` here?

Comment (biomadeira) [inline Bio/PDB/StructureBuilder.py]: @peterjc It doesn't make much difference, as we are initiating self.atom anyway in that 
`atom = self.atom = Atom()`
But I think I changed it by looking at the style used in init_chain() and init_residue() where the self.chain/residue were used...

Comment (peterjc) [inline Bio/PDB/StructureBuilder.py]: Sorry, yes - it was already exposed as ``.atom``. My mistake.

The old version just used the local variable ``atom`` as an alias, which would be very slightly faster than ``self.atom``.

**Comment by peterjc** [comment]
@JoaoRodrigues are you happy to handle this PR?

**Comment by JoaoRodrigues** [comment]
Sounds good to me, I'm testing it now on a few cases here.

Em qui, 2 de abr de 2015 às 11:03, Peter Cock <notifications@github.com>
escreveu:

**Comment by JoaoRodrigues** [comment]
Good to go on my side.

**Comment by peterjc** [comment]
Thank you both - merged to master. See also https://github.com/biopython/biopython/commit/b31c789c903ed157a0e0ff545b424500a9d15591

**Comment by biomadeira** [comment]
No problem and thanks to you both ;)

**Pull request merged**